src/frequenz/client/electricity_trading/_client.py

```diff
@@ -112,6 +117,7 @@ def __init__(self, grpc_channel: grpc.aio.Channel) -> None:
                 electricity_trading_pb2.ReceivePublicTradesStreamResponse, PublicTrade
             ],
         ] = {}
+        self._metadata = (("key", key),) if key else ()
```

Comment:
This is not sufficient to cover key based authorization. I suggest to remove this line and add all the necessary things in a follow up PR.